src/Components/Publishing/Sections/ViewFullscreen.tsx

```diff
@@ -3,52 +3,63 @@ import * as React from "react"
 import styled from "styled-components"
 import { track } from "../../../Utils/track"
 import IconExpand from "../Icon/Expand"
+import FullscreenViewer from "./FullscreenViewer/FullscreenViewer"
 
 interface ViewFullscreenProps extends React.HTMLProps<HTMLDivElement> {
   index?: number
+  images?: [object]
+  slideIndex?: number
 }
 
 @track()
-class ViewFullscreen extends React.Component<ViewFullscreenProps, void> {
+class ViewFullscreen extends React.Component<ViewFullscreenProps, any> {
   static childContextTypes = {
     onViewFullscreen: PropTypes.func,
   }
 
   constructor(props) {
     super()
-    this.onClick = this.onClick.bind(this)
+    this.openViewer = this.openViewer.bind(this)
+    this.state = { viewerIsOpen: false }
   }
 
   @track({ action: "Clicked article impression" })
-  onClick(e) {
+  openViewer(e) {
     e.preventDefault()
-    if (this.context.onViewFullscreen) {
-      this.context.onViewFullscreen(this.props.index)
-    }
+    this.setState({ viewerIsOpen: true })
+  }
+
+  closeViewer = e => {
+    this.setState({ viewerIsOpen: false })
   }
 
   render() {
     return (
-      <ViewFullscreenLink onClick={this.onClick}>
-        <IconExpand />
-      </ViewFullscreenLink>
+      <div>
+        <ViewFullscreenLink onClick={this.openViewer}>
+          <IconExpand />
+        </ViewFullscreenLink>
+        <FullscreenViewer
+          slideIndex={this.props.slideIndex}
+          onClose={this.closeViewer}
+          show={this.state.viewerIsOpen}
+          images={this.props.images}
+        />
```

Comment:
Doesn't this bloat the page _a lot_?

Reply:
IMO it's probably a premature optimization to consider the HTML bloat and something to be addressed _if_ there are performance problems after trying the most straightforward implementation. That said, it won't actually bloat the HTML size because if `show={false}` then it [won't render any HTML and instead return null](https://github.com/artsy/reaction/blob/master/src/Components/Publishing/Sections/FullscreenViewer/FullscreenViewer.tsx#L82).